src/Symfony/Component/AstGenerator/AstGeneratorInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AstGenerator;
+
+/**
+ * An AstGeneratorInterface is a contract to transform an object into an AST.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+interface AstGeneratorInterface
+{
+    /**
+     * Generate an object into an AST given a specific context.
+     *
+     * @param mixed $object  Object to generate AST from
+     * @param array $context Context for the generator
+     *
+     * @return \PhpParser\Node[] An array of statements (AST Node)
+     */
+    public function generate($object, array $context = array());
+
+    /**
+     * Check whether the given object is supported for generation by this generator.
+     *
+     * @param mixed $object Object to generate AST from
+     *
+     * @return bool
+     */
+    public function supportsGeneration($object);
```

Comment:
It is [useful for normalizer generation](https://github.com/joelwurtz/symfony/blob/8e636a098371584a1a49a81530036ee6cb6eeaff/src/Symfony/Component/AstGenerator/Hydrate/ObjectHydrateGenerator.php#L83-L94) you can take a look at [the small app i wrote](https://github.com/Ener-Getick/symfony-standard/blob/AST/src/AppBundle/Command/GeneratorCommand.php) to illustrate how #17516 could be used.